packages/@react-stately/table/src/useTableColumnResizeState.ts

```diff
@@ -46,8 +46,13 @@ export interface ColumnResizeStateProps<T> {
   tableWidth?: number
 }
 
-export function useTableColumnResizeState<T>(props: ColumnResizeStateProps<T>): ColumnResizeState<T> {
-  const {columns, getDefaultWidth, tableWidth: defaultTableWidth = null} = props;
+interface ColumnState<T> {
+  columns: GridNode<T>[]
+}
+
+export function useTableColumnResizeState<T>(props: ColumnResizeStateProps, state: ColumnState<T>): ColumnResizeState<T> {
```

Comment:
should the signature actually be 
```

interface Column {
  key: Key,
  defaultWidth: number,
  width: number,
  minWidth: number,
  maxWidth: number
}

interface ColumnState {
  columns: Column[]
}

export function useTableColumnResizeState<T>(props: ColumnResizeStateProps, state: ColumnState): ColumnResizeState<T> {
```

Reply:
Would this require extra changes to the TableCollection type definition? Wondering because we pass `state.collection` to this hook which has `columns: GridNode<T>[]`. Otherwise I guess we could construct an array of column data to match what you have above and pass that to `useTableColumnResizeState`, but what is the particular reason for the having the above interface?

Reply:
trying to decouple from our collections so someone could potentially use this with any table

Reply:
discussed with devon, i'll handle changing the interface in another PR